tests/integration_test.go

```diff
@@ -0,0 +1,90 @@
+package tests
+
+import (
+	"bytes"
+	"io/ioutil"
+	"path"
+	"path/filepath"
+	"testing"
+
+	skynet "github.com/NebulousLabs/go-skynet"
+	"gopkg.in/h2non/gock.v1"
+)
+
+// TestUploadAndDownloadFile tests uploading and downloading a single file.
+func TestUploadAndDownloadFile(t *testing.T) {
+	defer gock.Off() // Flush pending mocks after test execution
+
+	const srcFile = "../testdata/file1.txt"
+	const skylink = skynet.URISkynetPrefix + "testskynet"
+
+	// Test upload.
+
+	gock.New(skynet.DefaultUploadOptions.PortalURL).
+		Post(skynet.DefaultUploadOptions.PortalUploadPath).
+		Reply(200).
+		JSON(map[string]string{"skylink": "testskynet"})
+
+	skylink2, err := skynet.UploadFile(srcFile, skynet.DefaultUploadOptions)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if skylink2 != skylink {
+		t.Fatalf("expected skylink %v, got %v", skylink, skylink2)
+	}
+
+	// Test download.
+
+	gock.New(skynet.DefaultDownloadOptions.PortalURL).
+		Get(skynet.DefaultDownloadOptions.PortalDownloadPath).
+		Reply(200).
+		BodyString("test\n")
+
+	file, err := ioutil.TempFile("", t.Name())
+	if err != nil {
+		t.Fatal(err)
+	}
+	dstFile := file.Name()
+
+	err = skynet.DownloadFile(dstFile, skylink, skynet.DefaultDownloadOptions)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Check file equality.
+	f1, err1 := ioutil.ReadFile(srcFile)
+	if err1 != nil {
+		t.Fatal(err1)
+	}
+	f2, err2 := ioutil.ReadFile(path.Clean(dstFile))
+	if err2 != nil {
+		t.Fatal(err2)
+	}
+	if !bytes.Equal(f1, f2) {
```

Comment:
This is pretty neat but really `file1.txt` just contains the text "test". I wonder if `gock` is capable of doing more advanced stuff. I think we should look into it a bit more. I will do that as well to see what we could do to improve our testing coverage.